galsim/utilities.py

```diff
@@ -335,29 +462,36 @@ def thin_tabulated_values(x, f, rel_err=1.e-4, preserve_range=False):
     if total_integ == 0:
         return np.array([ x[0], x[-1] ]), np.array([ f[0], f[-1] ])
     thresh = rel_err * total_integ
+    x_range = x[-1] - x[0]
 
     if not preserve_range:
         # Remove values from the front that integrate to less than thresh.
-        integ = 0.5 * (abs(f[0]) + abs(f[1])) * (x[1] - x[0])
+        err_integ1 = 0.5 * (abs(f[0]) + abs(f[1])) * (x[1] - x[0])
         k0 = 0
-        while k0 < len(x)-2 and integ < thresh:
+        while k0 < len(x)-2 and err_integ1 < thresh * (x[k0+1]-x[0]) / x_range:
```

Comment:
I think this fix might be even better if the threshold were scaled by the cumulative integral up to x[k0+1], instead of just the x_range up to x[k0+1].  That way, it's still robust if someone prepends -1billion and 0 to x and f respectively.  I ran into this with the "official" LSST bandpasses, which all start at 300nm, even when the first few hundred entries all have throughput=0.